CodingTracker.AlvinGraham/CodingTracker.AlvinGraham/Utilities.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Spectre.Console;
+
+namespace CodingTracker;
+
+internal static class Utilities
+{
+
+	internal static void ClearScreen(string message)
+	{
+		Console.Clear();
+		Console.WriteLine(message);
+		Console.WriteLine("--------------------------------------");
+	}
+
+	internal static int GetNumber(string message)
+	{
+		string numberInput = AnsiConsole.Ask<string>(message);
+
+		if (numberInput == "0")
+			throw new ArgumentException("Returning to Main Menu");
+
+		var output = Validation.ValidateInt(numberInput, message);
+
+		return output;
+	}
+
+	internal static int GetGoalNumber(string message)
+	{
+		string numberInput = AnsiConsole.Ask<string>(message);
+
+		var output = Validation.ValidateInt(numberInput, message);
+
+		return output;
+	}
+
+	internal static DateTime[] GetDateInputs()
+	{
+		var startDateInput = AnsiConsole.Ask<string>("Input Start Date with the format: dd-mm-yy hh:mm (24 hour clock), or enter 0 to return to main menu.");
+
+		if (startDateInput == "0")
+			throw new ArgumentException("Returning to Main Menu");
```

Comment:
Throwing an Exception in these cases are not good practice.
Please update this in all places.
Try to display to user " You canceled the operation and Returning to Main Menu" return e.g. Null and update the code accordingly.

Reply:
Good call, thanks. I was trying to get cute with returning all the way to the MainMenu despite where it is in the call stack.  I see how this is bad practice and have refactored the code so that the method returns to the correct calling code and notifies the user as such.  I admit this was intellectual laziness on my part and was an easy fix.